Fix warnings from grep

c--- opened the pull request:
Newer versions of grep will issue a warning for invalid "escaped" slashes which in turn causes a bunch of warnings to appear while building.
```
grep (GNU grep) 3.8
grep: warning: stray \ before /
```

c--- commented:
This should probably be applied to openwrt-22.03 as well. Should I create another PR?

Ansuel commented on Makefile:
```diff
@@ -14,7 +14,7 @@ $(if $(findstring $(space),$(TOPDIR)),$(error ERROR: The path to the OpenWrt dir
 
 world:
 
-DISTRO_PKG_CONFIG:=$(shell $(TOPDIR)/scripts/command_all.sh pkg-config | grep -E '\/usr' -m 1)
```
Should we drop the `-E` ?

Ansuel commented on target/sdk/files/Makefile:
```diff
@@ -14,7 +14,7 @@ export TOPDIR LC_ALL LANG SDK
 
 world:
 
-DISTRO_PKG_CONFIG:=$(shell $(TOPDIR)/scripts/command_all.sh pkg-config | grep -E '\/usr' -m 1)
```
Should we drop the `-E` ?

httpstorm replied:
`-E` is for *extended regular expression* according to man, so we don't need it.

Ansuel replied:
yep that was the idea... no idea why it was used on the first place?

c--- commented:
@Ansuel What is that about markdown?

Hopefully I have split the commits correctly. Previously I thought maybe `-E` was reserved for future use but I guess there is no reason to keep it if will not be used. How does this look?

markmentovai commented:
See #10670.

httpstorm commented:
I performed some test builds on macOS on fresh git clone. Works fine.

Ansuel commented:
@httpstorm can you provide tested-by tag?

@c---  this part is not supported by git and should be removed or simply remove the ```
```
grep (GNU grep) 3.8
grep: warning: stray \ before /
```

c--- commented:
@Ansuel  Sorry, I thought I had removed that. I had to push an amend twice, hopefully it works out this time.

c--- committed:
build: fix warnings from grep

Fixes build warnings when using newer versions of grep.

Signed-off-by: Chris Osgood <chris_github@functionalfuture.com>
Tested-by: Georgi Valkov <gvalkov@gmail.com>

Ansuel commented:
@c--- can you rebase this and add the tested by tag?

Also the base-files commit should be dropped... The file is run under an openwrt system where busybox grep is used... Also this should be related to build warning not warning on running system.

c--- commented:
Merged with latest master and rebased the whole pull request. Added Tested-by.

Related to build warning? Is that a different indicator than `build:`?
